metaflow/plugins/kfp/kfp.py

```diff
@@ -96,48 +103,32 @@ def initial_setup_op_func(code_url: str)  -> StepOutput:
 
     print("\n----------RUNNING: METAFLOW INSTALLATION----------")
     subprocess.call(["pip3 install --user --upgrade {modified_metaflow_git_url}".format(
-        modified_metaflow_git_url=MODIFIED_METAFLOW_URL)],
-                    shell=True)
+        modified_metaflow_git_url=MODIFIED_METAFLOW_URL)], shell=True)
 
-    print("\n----------RUNNING: MAIN STEP COMMAND--------------")
+    print("\n----------RUNNING: INIT COMMAND-------------------")
     S3_BUCKET = os.getenv("S3_BUCKET")
     S3_AWS_ARN = os.getenv("S3_AWS_ARN")
     S3_AWS_REGION = os.getenv("S3_AWS_REGION")
 
     define_s3_env_vars = 'export METAFLOW_DATASTORE_SYSROOT_S3="{}" && export METAFLOW_AWS_ARN="{}" ' \
                          '&& export METAFLOW_AWS_S3_REGION="{}"'.format(S3_BUCKET, S3_AWS_ARN, S3_AWS_REGION)
     define_username = 'export USERNAME="kfp-user"'
-    python_cmd = 'python {0} --datastore="s3" --datastore-root="{1}" pre-start'.format(DEFAULT_DOWNLOADED_FLOW_FILENAME,
-                                                                                       S3_BUCKET)
-    final_run_cmd = f'{define_username} && {define_s3_env_vars} && {python_cmd}'
-
-    print("RUNNING COMMAND: ", final_run_cmd)
-    proc = subprocess.run(final_run_cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
-    proc_output = proc.stdout
-    proc_error = proc.stderr
+    init_cmd = 'python {0} --datastore="s3" --datastore-root="{1}" init --run-id={2} --task-id=0'.format(DEFAULT_DOWNLOADED_FLOW_FILENAME,
+                                                                                                           S3_BUCKET, run_id)
+    final_init_cmd = f'{define_username} && {define_s3_env_vars} && {init_cmd}'
 
-    StepOutput = namedtuple('StepOutput',
-                             ['ds_root', 'run_id'])
+    print("RUNNING COMMAND: ", final_init_cmd)
+    execute(final_init_cmd)
 
-    if len(proc_error) > 1:
-        print("_____________ STDERR:_____________________________")
-        print(proc_error)
-
-    if len(proc_output) > 1:
-        print("______________ STDOUT:____________________________")
-        print(proc_output)
-
-        # Read in the outputs (in the penultimate line) containing args needed for the next steps.
-        # Note: Output format is: ['ds_root', 'run_id']
-        outputs = (proc_output.split("\n")[-2]).split() # this contains the args needed for next steps to run
-
-    else:
-        raise RuntimeWarning("This step did not generate the correct args for next step to run. This might disrupt the workflow")
+    print("\n----------RUNNING: MAIN STEP COMMAND----------------")
+    start_cmd = start_command_template.format(ds_root=S3_BUCKET, run_id=run_id)
+    final_run_cmd = f'{define_username} && {define_s3_env_vars} && {start_cmd}'
```

Comment:
can you use f strings? that would fail python2.

Reply:
Done